validator/src/domain/event.rs

```diff
@@ -0,0 +1,244 @@
+use std::collections::HashMap;
+
+use chrono::{DateTime, Utc};
+use num_traits::CheckedSub;
+use serde::{Deserialize, Serialize};
+
+use domain::balances_map::get_balances_after_fees_tree;
+use domain::validator::message::Accounting;
+use domain::{AdUnit, BalancesMap, BigNum, Channel, ChannelId, DomainError};
+
+#[serde(tag = "type", rename_all = "SCREAMING_SNAKE_CASE")]
+#[derive(Serialize, Deserialize)]
+pub enum Event {
+    #[serde(rename_all = "camelCase")]
+    Impression {
+        publisher: String,
+        // clippy warning for big size difference, because of field
+        ad_unit: Box<AdUnit>,
```

Comment:
noo that should be an ID

Reply:
and it should be Option<_>

Reply:
> noo that should be an ID

Is this the ipfs hash ( the ID ) or something else? This is purely for me, as I still don't know everything and it was even hard to find the structures for the events.